types/dom-chromium-ai/index.d.ts

```diff
@@ -0,0 +1,253 @@
+interface WindowOrWorkerGlobalScope {
+    readonly ai: AI;
+}
+
+interface AI {
+    readonly assistant: AIAssistantFactory;
+    readonly summarizer: AISummarizerFactory;
+    readonly writer: AIWriterFactory;
+    readonly rewriter: AIRewriterFactory;
+}
+
+interface AICreateMonitor {
+    ondownloadprogress: (event: DownloadProgressEvent) => void;
+
+    addEventListener(
+        type: "downloadprogress",
+        listener: (event: DownloadProgressEvent) => void,
+        options?: AddEventListenerOptions | boolean,
+    ): void;
+    removeEventListener(
+        type: "downloadprogress",
+        listener: (event: DownloadProgressEvent) => void,
+        options?: EventListenerOptions | boolean,
+    ): void;
+}
+
+interface DownloadProgressEvent extends Event {
+    readonly loaded: number;
+    readonly total: number;
+}
+
+type AICreateMonitorCallback = (monitor: AICreateMonitor) => void;
+
+type AICapabilityAvailability = "readily" | "after-download" | "no";
+
+interface AIReadableStream extends ReadableStream<string> {
+    [Symbol.asyncIterator](): AsyncIterableIterator<string>;
+}
+
+// Assistant
+// https://github.com/explainers-by-googlers/prompt-api/#full-api-surface-in-web-idl
+
+interface AIAssistantFactory {
+    create(options?: AIAssistantCreateOptions): Promise<AIAssistant>;
+    capabilities(): Promise<AIAssistantCapabilities>;
+}
+
+interface AIAssistantCreateOptions {
+    signal?: AbortSignal;
+    monitor?: AICreateMonitorCallback;
+
+    systemPrompt?: string;
+    initialPrompts?: AIAssistantPrompt[];
+    topK?: number;
+    temperature?: number;
```

Comment:
The `temperature` in practice is capped by [`maxTopK](https://github.com/explainers-by-googlers/prompt-api/#:~:text=unsigned%20long%3F-,maxTopK,-%3B%0A%20%20readonly%20attribute%20float). Not sure if this is worth expressing (somehow?) in TypeScript syntax.

Reply:
I don't think this can be enforced by typings.